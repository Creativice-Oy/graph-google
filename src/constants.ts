import {
  StepEntityMetadata,
  RelationshipClass,
  StepRelationshipMetadata,
} from '@jupiterone/integration-sdk-core';

type EntityConstantKeys =
  | 'DOMAIN'
  | 'USER'
  | 'GROUP'
  | 'ACCOUNT'
  | 'SITE'
  | 'TOKEN';

export const entities: Record<EntityConstantKeys, StepEntityMetadata> = {
  DOMAIN: {
    resourceName: 'Domain',
    _type: 'google_domain',
    _class: 'Domain',
  },
  USER: {
    resourceName: 'User',
    _type: 'google_user',
    _class: 'User',
  },
  GROUP: {
    resourceName: 'Group',
    _type: 'google_group',
    _class: 'UserGroup',
  },
  ACCOUNT: {
    resourceName: 'Account',
    _type: 'google_account',
    _class: 'Account',
  },
  SITE: {
    resourceName: 'Site',
    _type: 'google_site',
    _class: 'Site',
  },
  TOKEN: {
    resourceName: 'Token',
    _type: 'google_token',
    _class: 'AccessKey',
  },
};

type RelationshipConstantKeys =
  | 'ACCOUNT_HAS_USER'
  | 'SITE_HOSTS_USER'
  | 'GROUP_HAS_USER'
  | 'GROUP_HAS_GROUP'
  | 'ACCOUNT_HAS_GROUP'
  | 'USER_ASSIGNED_TOKEN';

export const relationships: Record<
  RelationshipConstantKeys,
  StepRelationshipMetadata
> = {
  ACCOUNT_HAS_USER: {
    _type: 'google_account_has_user',
    _class: RelationshipClass.HAS,
    sourceType: entities.ACCOUNT._type,
    targetType: entities.USER._type,
  },
  SITE_HOSTS_USER: {
    _type: 'google_site_has_user',
    // TODO: Change to HOSTS
    _class: RelationshipClass.HAS,
    sourceType: entities.SITE._type,
    targetType: entities.USER._type,
  },
  GROUP_HAS_USER: {
    _type: 'google_group_has_user',
    _class: RelationshipClass.HAS,
    sourceType: entities.GROUP._type,
    targetType: entities.USER._type,
  },
  GROUP_HAS_GROUP: {
    _type: 'google_group_has_group',
    _class: RelationshipClass.HAS,
    sourceType: entities.GROUP._type,
    targetType: entities.GROUP._type,
  },
  ACCOUNT_HAS_GROUP: {
    _type: 'google_account_has_group',
    _class: RelationshipClass.HAS,
    sourceType: entities.ACCOUNT._type,
    targetType: entities.GROUP._type,
  },
  USER_ASSIGNED_TOKEN: {
    _type: 'google_user_assigned_token',
    _class: RelationshipClass.ASSIGNED,
    sourceType: entities.USER._type,
    targetType: entities.TOKEN._type,
  },
};
