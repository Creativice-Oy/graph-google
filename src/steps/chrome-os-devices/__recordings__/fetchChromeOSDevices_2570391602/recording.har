{
  "log": {
    "_recordingName": "fetchChromeOSDevices",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "5.1.1"
    },
    "entries": [
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 835,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/x-www-form-urlencoded"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "835"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 300,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/x-www-form-urlencoded",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 586,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 586,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "pragma",
              "value": "no-cache"
            },
            {
              "name": "date",
              "value": "Thu, 19 May 2022 17:43:06 GMT"
            },
            {
              "name": "cache-control",
              "value": "no-cache, no-store, max-age=0, must-revalidate"
            },
            {
              "name": "expires",
              "value": "Mon, 01 Jan 1990 00:00:00 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 603,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-05-19T17:43:06.782Z",
        "time": 177,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 177
        }
      },
      {
        "_id": "e2d5a379d8cfedf513936b98366dc85a",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/5.0.5 gl-node/14.17.0 auth/7.10.3"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/5.0.5 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "admin.googleapis.com"
            }
          ],
          "headersSize": 540,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://admin.googleapis.com/admin/directory/v1/customer/C048tgq5f/devices/chromeos"
        },
        "response": {
          "bodySize": 4162,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 4162,
            "text": "{\"kind\":\"admin#directory#chromeosdevices\",\"etag\":\"\\\"AX6M7SfKZ_nsxN3gDLY5FqkOIZdXe4fOIyur1mUmk1Y/qO5MJ7cykk0pduKRH2T6XZtT4pg\\\"\",\"chromeosdevices\":[{\"kind\":\"admin#directory#chromeosdevice\",\"etag\":\"\\\"AX6M7SfKZ_nsxN3gDLY5FqkOIZdXe4fOIyur1mUmk1Y/_JUms9o7An2qTIl3Rci_r7xsAEo\\\"\",\"deviceId\":\"06d4fd60-bed5-4fe2-8d82-1a7b7b76f9ac\",\"serialNumber\":\"3WAZ9FGNC00030E\",\"status\":\"ACTIVE\",\"lastSync\":\"2022-05-19T16:14:15.759Z\",\"annotatedUser\":\"integration-development@jupiterone.dev\",\"model\":\"Samsung Galaxy Chromebook\",\"osVersion\":\"101.0.4951.59\",\"platformVersion\":\"14588.98.0 (Official Build) stable-channel hatch\",\"firmwareVersion\":\"Google_Kohaku.12672.463.0\",\"macAddress\":\"2cdb07af9a83\",\"bootMode\":\"Verified\",\"lastEnrollmentTime\":\"2022-05-19T13:33:02.539Z\",\"orgUnitPath\":\"/\",\"orgUnitId\":\"03ph8a2z2lubvlt\",\"activeTimeRanges\":[{\"date\":\"2022-05-19\",\"activeTime\":840024}],\"tpmVersionInfo\":{\"family\":\"322e3000\",\"specLevel\":\"74\",\"manufacturer\":\"43524f53\",\"tpmModel\":\"1\",\"firmwareVersion\":\"a2d932d70cb1415b\",\"vendorSpecific\":\"xCG fTPM\"},\"cpuStatusReports\":[{\"reportTime\":\"2022-05-19T13:33:09.569Z\",\"cpuUtilizationPercentageInfo\":[1],\"cpuTemperatureInfo\":[{\"temperature\":33,\"label\":\"Core 0\\n\"},{\"temperature\":35,\"label\":\"Core 1\\n\"},{\"temperature\":33,\"label\":\"Core 2\\n\"},{\"temperature\":32,\"label\":\"Core 3\\n\"},{\"temperature\":35,\"label\":\"Package id 0\\n\"},{\"temperature\":30,\"label\":\"iwlwifi_1\\n\"}]},{\"reportTime\":\"2022-05-19T13:34:09.558Z\",\"cpuTemperatureInfo\":[{\"temperature\":29,\"label\":\"Core 0\\n\"},{\"temperature\":33,\"label\":\"Core 1\\n\"},{\"temperature\":33,\"label\":\"Core 2\\n\"},{\"temperature\":28,\"label\":\"Core 3\\n\"},{\"temperature\":33,\"label\":\"Package id 0\\n\"},{\"temperature\":30,\"label\":\"iwlwifi_1\\n\"}]},{\"reportTime\":\"2022-05-19T14:03:13.172Z\",\"cpuUtilizationPercentageInfo\":[39]},{\"reportTime\":\"2022-05-19T14:04:13.166Z\",\"cpuTemperatureInfo\":[{\"temperature\":34,\"label\":\"Core 1\\n\"},{\"temperature\":32,\"label\":\"Core 2\\n\"},{\"temperature\":34,\"label\":\"Package id 0\\n\"},{\"temperature\":30,\"label\":\"Core 3\\n\"},{\"temperature\":31,\"label\":\"Core 0\\n\"},{\"temperature\":33,\"label\":\"iwlwifi_1\\n\"}]},{\"reportTime\":\"2022-05-19T16:14:13.428Z\",\"cpuUtilizationPercentageInfo\":[17]},{\"reportTime\":\"2022-05-19T16:15:13.424Z\",\"cpuTemperatureInfo\":[{\"temperature\":31,\"label\":\"Core 1\\n\"},{\"temperature\":28,\"label\":\"Core 2\\n\"},{\"temperature\":31,\"label\":\"Package id 0\\n\"},{\"temperature\":26,\"label\":\"Core 3\\n\"},{\"temperature\":28,\"label\":\"Core 0\\n\"},{\"temperature\":29,\"label\":\"iwlwifi_1\\n\"}]}],\"systemRamTotal\":\"8218046464\",\"systemRamFreeReports\":[{\"reportTime\":\"2022-05-19T13:33:09.569Z\",\"systemRamFreeInfo\":[\"5896839168\"]},{\"reportTime\":\"2022-05-19T14:03:13.162Z\",\"systemRamFreeInfo\":[\"6141267968\"]},{\"reportTime\":\"2022-05-19T16:14:13.420Z\",\"systemRamFreeInfo\":[\"6376738816\"]}],\"diskVolumeReports\":[{\"volumeInfo\":[{\"volumeId\":\"/media/archive\",\"storageTotal\":\"4109021184\",\"storageFree\":\"4109021184\"},{\"volumeId\":\"/usr/share/oem\",\"storageTotal\":\"12042240\",\"storageFree\":\"11681792\"},{\"volumeId\":\"/media/removable\",\"storageTotal\":\"4109021184\",\"storageFree\":\"4109021184\"}]}],\"manufactureDate\":\"2020-12-04\",\"autoUpdateExpiration\":\"1843455600000\",\"cpuInfo\":[{\"model\":\"Intel(R) Core(TM) i5-10210U CPU @ 1.60GHz\",\"architecture\":\"x64\",\"maxClockSpeedKhz\":4200000,\"logicalCpus\":[{\"maxScalingFrequencyKhz\":4200000,\"currentScalingFrequencyKhz\":887947,\"idleDuration\":\"72s\",\"cStates\":[{\"displayName\":\"C7s\",\"sessionDuration\":\"0.012074s\"},{\"displayName\":\"C3\",\"sessionDuration\":\"0.300112s\"},{\"displayName\":\"C1\",\"sessionDuration\":\"0.304366s\"},{\"displayName\":\"C10\",\"sessionDuration\":\"58.455156s\"},{\"displayName\":\"C8\",\"sessionDuration\":\"5.201451s\"},{\"displayName\":\"C6\",\"sessionDuration\":\"5.821382s\"},{\"displayName\":\"C1E\",\"sessionDuration\":\"0.748027s\"},{\"displayName\":\"POLL\",\"sessionDuration\":\"0.574729s\"},{\"displayName\":\"C9\",\"sessionDuration\":\"1.039921s\"}]},{\"maxScalingFrequencyKhz\":4200000,\"currentScalingFrequencyKhz\":889224,\"idleDuration\":\"72s\",\"cStates\":[{\"displayName\":\"C7s\",\"sessionDuration\":\"0.001092s\"},{\"displayName\":\"C3\",\"sessionDuration\":\"0.241435s\"},{\"displayName\":\"C1\",\"sessionDuration\":\"0.205430s\"},{\"displayName\":\"C10\",\"sessionDuration\":\"63.969573s\"},{\"displayName\":\"C8\",\"sessionDuration\":\"2.916627s\"},{\"displayName\":\"C6\",\"sessionDuration\":\"3.221630s\"},{\"displayName\":\"C1E\",\"sessionDuration\":\"0.403501s\"},{\"displayName\":\"POLL\",\"sessionDuration\":\"0.566464s\"},{\"displayName\":\"C9\",\"sessionDuration\":\"0.845658s\"}]},{\"maxScalingFrequencyKhz\":4200000,\"currentScalingFrequencyKhz\":887861,\"idleDuration\":\"71s\",\"cStates\":[{\"displayName\":\"C7s\",\"sessionDuration\":\"0.010559s\"},{\"displayName\":\"C3\",\"sessionDuration\":\"0.311136s\"},{\"displayName\":\"C1\",\"sessionDuration\":\"0.303459s\"},{\"displayName\":\"C10\",\"sessionDuration\":\"59.786169s\"},{\"displayName\":\"C8\",\"sessionDuration\":\"4.201012s\"},{\"displayName\":\"C6\",\"sessionDuration\":\"4.662529s\"},{\"displayName\":\"C1E\",\"sessionDuration\":\"0.771406s\"},{\"displayName\":\"POLL\",\"sessionDuration\":\"0.595660s\"},{\"displayName\":\"C9\",\"sessionDuration\":\"0.596215s\"}]},{\"maxScalingFrequencyKhz\":4200000,\"currentScalingFrequencyKhz\":909618,\"idleDuration\":\"67s\",\"cStates\":[{\"displayName\":\"C7s\",\"sessionDuration\":\"0.013031s\"},{\"displayName\":\"C3\",\"sessionDuration\":\"0.235805s\"},{\"displayName\":\"C1\",\"sessionDuration\":\"0.755466s\"},{\"displayName\":\"C10\",\"sessionDuration\":\"58.822622s\"},{\"displayName\":\"C8\",\"sessionDuration\":\"3.581926s\"},{\"displayName\":\"C6\",\"sessionDuration\":\"2.772003s\"},{\"displayName\":\"C1E\",\"sessionDuration\":\"0.515377s\"},{\"displayName\":\"POLL\",\"sessionDuration\":\"0.621053s\"},{\"displayName\":\"C9\",\"sessionDuration\":\"0.340984s\"}]},{\"maxScalingFrequencyKhz\":4200000,\"currentScalingFrequencyKhz\":874517,\"idleDuration\":\"72s\",\"cStates\":[{\"displayName\":\"C7s\",\"sessionDuration\":\"0.004804s\"},{\"displayName\":\"C3\",\"sessionDuration\":\"0.419223s\"},{\"displayName\":\"C1\",\"sessionDuration\":\"0.345607s\"},{\"displayName\":\"C10\",\"sessionDuration\":\"58.378115s\"},{\"displayName\":\"C8\",\"sessionDuration\":\"4.807022s\"},{\"displayName\":\"C6\",\"sessionDuration\":\"6.234130s\"},{\"displayName\":\"C1E\",\"sessionDuration\":\"1.003590s\"},{\"displayName\":\"POLL\",\"sessionDuration\":\"0.618255s\"},{\"displayName\":\"C9\",\"sessionDuration\":\"0.623138s\"}]},{\"maxScalingFrequencyKhz\":4200000,\"currentScalingFrequencyKhz\":899925,\"idleDuration\":\"73s\",\"cStates\":[{\"displayName\":\"C7s\",\"sessionDuration\":\"0.012425s\"},{\"displayName\":\"C3\",\"sessionDuration\":\"0.240779s\"},{\"displayName\":\"C1\",\"sessionDuration\":\"1.035389s\"},{\"displayName\":\"C10\",\"sessionDuration\":\"62.900368s\"},{\"displayName\":\"C8\",\"sessionDuration\":\"3.325270s\"},{\"displayName\":\"C6\",\"sessionDuration\":\"3.529959s\"},{\"displayName\":\"C1E\",\"sessionDuration\":\"0.862708s\"},{\"displayName\":\"POLL\",\"sessionDuration\":\"0.606138s\"},{\"displayName\":\"C9\",\"sessionDuration\":\"0.397835s\"}]},{\"maxScalingFrequencyKhz\":4200000,\"currentScalingFrequencyKhz\":862124,\"idleDuration\":\"72s\",\"cStates\":[{\"displayName\":\"C7s\",\"sessionDuration\":\"0.009301s\"},{\"displayName\":\"C3\",\"sessionDuration\":\"0.339881s\"},{\"displayName\":\"C1\",\"sessionDuration\":\"0.301033s\"},{\"displayName\":\"C10\",\"sessionDuration\":\"59.498871s\"},{\"displayName\":\"C8\",\"sessionDuration\":\"5.115185s\"},{\"displayName\":\"C6\",\"sessionDuration\":\"5.505044s\"},{\"displayName\":\"C1E\",\"sessionDuration\":\"0.776398s\"},{\"displayName\":\"POLL\",\"sessionDuration\":\"0.567139s\"},{\"displayName\":\"C9\",\"sessionDuration\":\"0.124393s\"}]},{\"maxScalingFrequencyKhz\":4200000,\"currentScalingFrequencyKhz\":825840,\"idleDuration\":\"73s\",\"cStates\":[{\"displayName\":\"C7s\",\"sessionDuration\":\"0.004143s\"},{\"displayName\":\"C3\",\"sessionDuration\":\"0.278739s\"},{\"displayName\":\"C1\",\"sessionDuration\":\"0.249963s\"},{\"displayName\":\"C10\",\"sessionDuration\":\"64.623793s\"},{\"displayName\":\"C8\",\"sessionDuration\":\"2.857472s\"},{\"displayName\":\"C6\",\"sessionDuration\":\"3.865774s\"},{\"displayName\":\"C1E\",\"sessionDuration\":\"0.486912s\"},{\"displayName\":\"POLL\",\"sessionDuration\":\"0.575661s\"},{\"displayName\":\"C9\",\"sessionDuration\":\"0.461379s\"}]}]}]}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "etag",
              "value": "\"AX6M7SfKZ_nsxN3gDLY5FqkOIZdXe4fOIyur1mUmk1Y/qO5MJ7cykk0pduKRH2T6XZtT4pg\""
            },
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Thu, 19 May 2022 17:43:07 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 564,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-05-19T17:43:06.969Z",
        "time": 394,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 394
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
