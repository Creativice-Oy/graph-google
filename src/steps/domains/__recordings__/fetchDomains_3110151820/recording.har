{
  "log": {
    "_recordingName": "fetchDomains",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 977,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "977"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 283,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/json",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 592,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 592,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Thu, 27 Aug 2020 18:40:46 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-27=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-T050=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 563,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2020-08-27T18:40:46.678Z",
        "time": 174,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 174
        }
      },
      {
        "_id": "81708a4628a0f4034553d9ea4fc24931",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/4.4.0 gl-node/13.10.1 auth/6.0.6"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/4.4.0 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 538,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://www.googleapis.com/admin/directory/v1/customer/C02etu699/domains"
        },
        "response": {
          "bodySize": 975,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 975,
            "text": "{\"kind\":\"admin#directory#domains\",\"etag\":\"\\\"o6ztWZ3lhe-tsV75FsFXpSEZSr0dIx_rEPuaEFoI634/KTGhUSwA2KNfLDZOYUFxZfciSzM\\\"\",\"domains\":[{\"kind\":\"admin#directory#domain\",\"etag\":\"\\\"o6ztWZ3lhe-tsV75FsFXpSEZSr0dIx_rEPuaEFoI634/6j1NZUK3BBPEtHB3FKBAuzFYmEk\\\"\",\"domainName\":\"jupiterone.io\",\"domainAliases\":[{\"kind\":\"admin#directory#domainAlias\",\"etag\":\"\\\"o6ztWZ3lhe-tsV75FsFXpSEZSr0dIx_rEPuaEFoI634/zevudprFOEPK3lTezAE-d43Sy3o\\\"\",\"domainAliasName\":\"jupiterone.io.test-google-a.com\",\"parentDomainName\":\"jupiterone.io\",\"verified\":true,\"creationTime\":\"1549555037874\"}],\"isPrimary\":false,\"verified\":true,\"creationTime\":\"1549555037874\"},{\"kind\":\"admin#directory#domain\",\"etag\":\"\\\"o6ztWZ3lhe-tsV75FsFXpSEZSr0dIx_rEPuaEFoI634/CdS83q1_xnaOGbB-PXkIZoldDVM\\\"\",\"domainName\":\"jupiterone.com\",\"isPrimary\":true,\"verified\":true,\"creationTime\":\"1594817987271\"},{\"kind\":\"admin#directory#domain\",\"etag\":\"\\\"o6ztWZ3lhe-tsV75FsFXpSEZSr0dIx_rEPuaEFoI634/2dHzznB4s2jx9W_p2IM6KWdb0rQ\\\"\",\"domainName\":\"jptr.one\",\"isPrimary\":false,\"verified\":true,\"creationTime\":\"1596544251506\"}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "etag",
              "value": "\"o6ztWZ3lhe-tsV75FsFXpSEZSr0dIx_rEPuaEFoI634/KTGhUSwA2KNfLDZOYUFxZfciSzM\""
            },
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Thu, 27 Aug 2020 18:40:47 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-27=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-T050=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 621,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2020-08-27T18:40:46.867Z",
        "time": 242,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 242
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
