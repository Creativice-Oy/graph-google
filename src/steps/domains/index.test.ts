import {
  Recording,
  createMockStepExecutionContext,
} from '@jupiterone/integration-sdk-testing';
import { setupIntegrationRecording } from '../../../test/recording';
import { IntegrationConfig } from '../../types';
import { fetchDomains } from '.';
import { integrationConfig } from '../../../test/config';
import { entities } from '../../constants';

describe('#fetchDomains', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupIntegrationRecording({
      directory: __dirname,
      name: 'fetchDomains',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchDomains(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(context.jobState.collectedEntities).toMatchGraphObjectSchema({
      _class: [entities.DOMAIN._class],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: entities.DOMAIN._type },
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          emailDomain: { type: 'string' },
          primary: { type: 'boolean' },
          verified: { type: 'boolean' },
        },
      },
    });
  });
});
