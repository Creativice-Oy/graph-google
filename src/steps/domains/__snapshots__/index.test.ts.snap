// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchDomains should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Domain",
      ],
      "_key": "google_domain_jupiterone.dev",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "creationTime": "1622061210453",
            "domainAliases": Array [
              Object {
                "creationTime": "1622061210453",
                "domainAliasName": "jupiterone.dev.test-google-a.com",
                "etag": "\\"oUFCDjWEzoqbOg-UBcrPNLm0LDKHeK9dytjyFPUxPDA/oW480BdgHayQKmnuoDeWTf6fPjs\\"",
                "kind": "admin#directory#domainAlias",
                "parentDomainName": "jupiterone.dev",
                "verified": true,
              },
            ],
            "domainName": "jupiterone.dev",
            "etag": "\\"oUFCDjWEzoqbOg-UBcrPNLm0LDKHeK9dytjyFPUxPDA/WA7ByZRkRpGgnXLwEPGUupV7k6s\\"",
            "isPrimary": true,
            "kind": "admin#directory#domain",
            "verified": true,
          },
        },
      ],
      "_type": "google_domain",
      "createdOn": 1622061210453,
      "displayName": "jupiterone.dev",
      "domainName": "jupiterone.dev",
      "emailDomain": "jupiterone.dev",
      "id": "jupiterone.dev",
      "name": "jupiterone.dev",
      "primary": true,
      "verified": true,
    },
  ],
  "collectedRelationships": Array [],
  "encounteredTypes": Array [
    "google_domain",
  ],
  "numCollectedEntities": 1,
  "numCollectedRelationships": 0,
}
`;
